packages/opentelemetry-sdk-trace-base/test/common/Span.test.ts

```diff
@@ -1203,6 +1203,46 @@ describe('Span', () => {
         const event = span.events[0];
         assert.deepStrictEqual(event.time, [0, 123]);
       });
+
+      it('should record an exception with provided time as a 3rd arg', () => {
+        const span = new Span(
+          tracer,
+          ROOT_CONTEXT,
+          name,
+          spanContext,
+          SpanKind.CLIENT
+        );
+        // @ts-expect-error writing readonly property. performance time origin is mocked to return ms value of [1,1]
+        span['_performanceOffset'] = 0;
+        assert.strictEqual(span.events.length, 0);
+        span.recordException('boom', undefined, [0, 123]);
+        const event = span.events[0];
+        assert.deepStrictEqual(event.time, [0, 123]);
+      });
+    });
+
+    describe('when attributes are provided', () => {
+      it('should sanitized and merge attributes when provided', () => {
```

Comment:
According to the spec " If attributes with the same name would be generated by the method already, the additional attributes take precedence."  it would be good to add test specific for this.

Reply:
Excellent catch! I updated the implementation and added the tests. PTAL.